src/polysolve/linear/TrilinosSolver.cpp

```diff
@@ -0,0 +1,260 @@
+#ifdef POLYSOLVE_WITH_TRILINOS
+
+////////////////////////////////////////////////////////////////////////////////
+#include "TrilinosSolver.hpp"
+#include <string>
+#include <vector>
+#include <unsupported/Eigen/SparseExtra>
+////////////////////////////////////////////////////////////////////////////////
+
+Eigen::MatrixXd test_vertices;
+Eigen::MatrixXd init_vertices;
+std::vector<int> test_boundary_nodes;
+Eigen::MatrixXd remove_boundary_vertices(const Eigen::MatrixXd &vertices, const std::vector<int> &boundary_nodes)
+{
+    // Remove boundary vertices
+    if (boundary_nodes.empty())
+    {
+        return vertices;
+    }
+    else
+    {
+        std::vector<int> order_nodes = boundary_nodes;
+        std::sort(order_nodes.begin(), order_nodes.end());
+        Eigen::MatrixXd out_vertices;
+        std::vector<int> keep;
+        for (int i = 0; i < vertices.rows(); i++)
+        {
+            if (!std::binary_search(order_nodes.begin(), order_nodes.end(),i))
+            {
+                keep.push_back(i);
+            }
+        }
+        out_vertices = vertices(keep, Eigen::all);
+        return out_vertices;
+    }
+}
+
+namespace polysolve::linear
+{
+    TrilinosSolver::TrilinosSolver()
+    {
+        precond_num_ = 0;
+// #ifdef HAVE_MPI
+        int done_already;
+        MPI_Initialized(&done_already);
+        if (!done_already)
+        {
+            /* Initialize MPI */
+            int argc = 1;
+            char name[] = "";
+            char *argv[] = {name};
+            char **argvv = &argv[0];
+            MPI_Init(&argc, &argvv);
+            CommPtr = new Epetra_MpiComm(MPI_COMM_WORLD);
+        }
+// #else
+    //  CommPtr=new Epetra_SerialComm;
+// #endif
+    }
+
+    ////////////////////////////////////////////////////////////////
+    void TrilinosSolver::set_parameters(const json &params)
+    {
+        if (params.contains("Trilinos"))
+        {
+            if (params["Trilinos"].contains("block_size"))
+            {
+                if (params["Trilinos"]["block_size"] == 2 || params["Trilinos"]["block_size"] == 3)
+                {
+                    numPDEs = params["Trilinos"]["block_size"];
+                }
+            }
+            if (params["Trilinos"].contains("max_iter"))
+            {
+                max_iter_ = params["Trilinos"]["max_iter"];
+            }
+            if (params["Trilinos"].contains("tolerance"))
+            {
+                conv_tol_ = params["Trilinos"]["tolerance"];
+            }
+            if (params["Trilinos"].contains("is_nullspace"))
```

Comment:
this is also a method

Reply:
is_nullspace is not necessary to exist anymore, so it is removed.